### PR TITLE
Fix mbp emptyrepo name

### PR DESCRIPTION
**Pull request by vivek**
# Description

- Fixes bug where Editor dialog page shows title as 'reponame/undefined' for multi-branch pipeline with empty repo. Fix is to show 'reponame'. 
- Transition to activity page after receiving MBP indexing sse completion event with a timeout

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: src/main/js/SaveApi.js [hunk 1]
```diff
@@ -33,36 +56,34 @@ export class SaveApi {
...
-        const sseId = sseService.registerHandler(event => {
-            if (event.job_multibranch_indexing_result === 'SUCCESS') {
-                cleanup();
-            }
-            if (event.job_multibranch_indexing_result === 'FAILURE') {
-                cleanup({ message: 'Indexing failed' });
-            }
-        });
-
+            cleanup(timeoutId, complete, onError);
```
Comment (kzantow) [inline src/main/js/SaveApi.js]: looks like `cleanup` was removed; should this be `this._cleanup` ?

File: src/main/js/SaveApi.js [hunk 2]
```diff
@@ -1,10 +1,33 @@
...
 export class SaveApi {
 
+    _cleanup(sseId, timeoutId, onComplete, onError, err) {
+        sseService.removeHandler(sseId);
+        clearTimeout(timeoutId);
+        loadingIndicator.hide();
+        if (err && onError) {
+            onError(err);
+        } else {
+            onComplete();
```
Comment (kzantow) [inline src/main/js/SaveApi.js]: also looks like `onComplete` will be called twice 🐛

Comment (vivek) [inline src/main/js/SaveApi.js]: fixed

**Comment by vivek** [comment]
@kzantow Fixed issues pointed in your review. ATH and manual testing looks good. PTAL.

**Pull request merged**